po4a: update to 0.66

newbluemoon opened the pull request:
Two sgml tests fail because they invoke `onsgmls` from OpenSP to verify a test sgml file against a docbook DTD which errors out.

The `opensp` package adds a sgml catalog entry
`CATALOG /usr/share/OpenSP/catalog`
and in `/usr/share/OpenSP/catalog` there is a sgml declaration:
`SGMLDECL unicode.sd`

When this line or the catalog entry above is removed `onsgmls` falls back to its
default definition and succeeds and all tests are passed.

I’m not sure if the problem is related to Void’s `opensp` package or to `po4a`.
Any ideas are welcome! :)
I also asked upstream: https://github.com/mquinson/po4a/issues/327

<!-- Uncomment relevant sections and delete options which are not applicable -->

#### Testing the changes
- I tested the changes in this PR: **NO**

<!--
#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): **YES**|**NO**
-->

<!-- Note: If the build is likely to take more than 2 hours, please [skip CI](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration)
and test at least one native build and, if supported, at least one cross build.
Ignore this section if this PR is not skipping CI.
-->
<!-- 
#### Local build testing
- I built this PR locally for my native architecture, (ARCH-LIBC)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64-musl
  - armv7l
  - armv6l-musl
-->

ericonr commented:
We build `opensp` with `--enable-default-catalog=/usr/share/sgml/catalog`, could that affect things? We are also one version above the one advertised in the homepage,

newbluemoon commented:
`opensp` without `--enable-default-catalog=...` doesn’t find the docbook DTDs in the first place. :(

All distributions I checked use 1.5.2. Maybe they just forgot or didn’t care to update the homepage. ;)

What’s bugging me is why does
`onsgmls -p basic.sgml` work and
`onsgmls -p < basic.sgml` doesn’t.
(`basic.sgml` from `po4a-0.64/f/fmt/sgml/`)

http://openjade.sourceforge.net/doc/nsgmls.htm says
`onsgmls [ option... ] sysid...`

> If no system identifiers are specified, then nsgmls will read the document entity from the standard input. A command line system identifier of - can be used to refer to the standard input. (Normally in a system identifier, <osfd>0 is used to refer to standard input.)


http://openjade.sourceforge.net/doc/sysid.htm says
> There are two kinds of system identifier: formal system identifiers and simple system identifiers. A system identifier that does not start with < will always be interpreted as a simple system identifier. A simple system identifier will always be interpreted either as a filename or as a URL.
...
A system identifier will be interpreted as a formal system identifier if it starts with a < followed by a storage manager name, followed by either > or white-space; otherwise it will be interpreted as a simple system identifier.

But in the above example the `<` is a shell redirection not the start of a formal system identifier so that the file is supposed to be read via stdin. I can only assume that in this case “something” gets lost which is expected to also be fed in via stdin?

newbluemoon commented:
Added an upstream patch which removes piping into `onsgmls`, reading the file as a paramater instead. This seems to have been a residual from a long time ago and fixes half of the problem.

The test still fails because `docbook` ships a declaration which has `OMITTAG NO`. When set to `YES` the test passes. Debian for example completely patches out all `docbook` declarations and ships its own.

This seems to happen because the test sgml files are not valid according to the original docbook declaration and the way to go should be to fix the test sgml files. I will give it a shot, but until then, I think this PR is good to go as it is the test that’s broken. :)

newbluemoon commented:
Backported two upstream patches to fix the sgml tests.
All should be fine now. :)

newbluemoon commented:
The sgml-test is fixed and passes, but now some other failures show up...
Tests in local builds pass on x86_64 and i686, fail, however, on x86_64-musl.

Temporarily enabling verbose test output for the `perl-ModuleBuild`-build-style to see where CI fails.

newbluemoon commented:
The errors on x86_64-musl like e.g.
```
# FAILED command: PODIFF  -I#:  cfg/multiple-nopotpo/_multiple.de.po tmp/cfg/multiple-nopotpo-src/multiple.de.po 2>&1 > tmp/cfg/multiple-nopotpo-src/_cmd_output
# Command output:
# | --- cfg/multiple-nopotpo/_multiple.de.po	2021-11-14 12:08:53.000000000 +0000
# | +++ tmp/cfg/multiple-nopotpo-src/multiple.de.po	2021-11-18 19:04:08.059796074 +0000
# | @@ -1,7 +1,8 @@
# |  # German translations for po package
# | -# Copyright (C) 2020 Free Software Foundation, Inc.
# | +# German translation for po
# | +# Copyright (C) 2021 Free Software Foundation, Inc.
# |  # This file is distributed under the same license as the po package.
# | -# Automatically generated, 2020.
# | +# Automatically generated, 2021.
# |  #
# |  msgid ""
# |  msgstr ""
# (end of output)
```

are caused because musl’s gettext seems to add a translation for the package description while gettext on glibc does not.
So that there is an additional line `# | +# German translation for po` in the example above which doesn’t exist in the original .po file with which the newly generated one is compared. It comes from `/usr/share/locale/*/LC_MESSAGES/gettext-tools.mo`. The other differences are excluded via `diff -I` and don’t matter.
No idea how to fix this right now, if it is worth fixing? @ericonr maybe you know something about the differences of gettext on musl and glibc? :)

All other CI errors are like 
```
Broken test cfg/split-multimaster: cfg/split-multimaster/_second.man.fr should be the expected content of produced file second.man.fr
```
and I cannot reproduce them locally.

ericonr commented:
> are caused because musl’s gettext seems to add a translation for the package description while gettext on glibc does not.

Which is amazing, given that they are both the same package ;-;

> and I cannot reproduce them locally.

Could be something fucky to do with running as root ;-;

If the tests complete successfully locally on glibc, I'd be okay with `make_check=ci-skip # can't reproduce failure locally`

newbluemoon commented:
Ran tests locally on x86_64 and i686 without any issue and added `make_check=ci-skip ...` as you suggested.
Also removed the temporary verbose test output for `perl-ModuleBuild`.
So I think now it should be really good to go. But I’ll still try to figure out where those strange test failures come from... :)

ericonr commented:
Sorry for the delay merging, and thanks for keeping on it :)

newbluemoon commented:
@ericonr No worries, good to see you’re back! :)
